src/Application/Query.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+final class Query {
```

Comment:
I assume all these `final`s were added by AI? Or are we explicitly preventing subclassing? (I'm not used to seeing this in our other code, even when classes are not really meant to be subclassed).

Reply:
Ah yes, indeed. I did not even notice those.

Reply:
_(Jeroen-o1 thought about the final keyword for 5 minutes)_

For our projects there seems little difference in having final slapped on most things. We rarely use protected visibility and almost never use inheritance for code reuse.

While my inner programmer wants to spend even more brain cycles on figuring out what the best policy is. However, productivity Jeroen is saying this is a waste of time. So shrug/whatever.

Reply:
Indeed doesn't really matter here. My feeling is to not add it, as we've done in other projects, unless there's a very explicit reason to do so (which I don't think we've ever had, hence why I noticed it).